.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - $default-branch
+  workflow_dispatch:
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Build with MkDocs
+        run: |
+          python -m pip install -r docs/requirements.txt
+          mkdocs build
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: "./site"
+
+  deploy:
+    needs: build
+
+    # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
+    permissions:
+      pages: write # to deploy to Pages
+      id-token: write # to verify the deployment originates from an appropriate source
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
It looks like we will need to configure carma-platform's settings to support deploying to the repository's GitHub Pages ([https://github.com/usdot-fhwa-stol/carma-platform/settings/pages](https://github.com/usdot-fhwa-stol/carma-platform/settings/pages)). Since this PR is just the skeleton, do you envision that being part of future work?

Reply:
It's one or two settings, we could do it after merging this PR. I don't think it's significant enough to formally track.